### ISSUE TITLE
could not convert BatchDescriptor to cudnn tensor descriptor: CUDNN_STATUS_BAD_PARAM

### ISSUE THREAD

**Issue by MrWanter** [ppwwyyxx/tensorpack#713]
Hi! Recently I came across a problem when using your batchnorm layer, I used it like 
```
from tensorpack.models.batch_norm import BatchNorm
BatchNorm('bn', x, data_format='NCHW')
```
but I got 
```
2018-03-28 17:14:56.970757: F tensorflow/stream_executor/cuda/cuda_dnn.cc:444] could not convert BatchDescriptor {count: 5 feature_map_count: 0 spatial: 24 78  value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX} to cudnn tensor descriptor: CUDNN_STATUS_BAD_PARAM
Aborted
```
I saw you have a bug PR for tensorflow https://github.com/tensorflow/tensorflow/issues/14657 on this problem, but I checked that I dont have 0-batch size input, I'm using tensorflow `1.4.1`, could you pls point out where possibly could be wrong?

**Comment by ppwwyyxx** [comment]
You inputs have zero channel. How are you supposed to do batch norm.
